File: Data Structures & Algorithms/last-stone-weight/submission-1.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        stones = [-x for x in stones]
        maxHeap = stones
        heapq.heapify(maxHeap)

        while len(maxHeap) > 1:
            first = -1 * heapq.heappop(maxHeap)
            second = -1 * heapq.heappop(maxHeap)
            res = first - second
            if res > 0:
                heapq.heappush(maxHeap, -1 * res)
        
        if len(maxHeap) == 1:
            return -1 * heapq.heappop(maxHeap)
        else:
            return 0

File: Data Structures & Algorithms/longest-common-prefix/submission-1.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(), strs.end());
        string first = strs[0];
        string second = strs[strs.size()-1];

        int n = min(first.size(),second.size());
        string res = "";

        for (int i=0; i<n; i++) {
            if (first[i] == second[i]) {
                res += first[i];
            } else {
                break;
            }
        }

        return res;


    }
};